java/build.gradle.kts

```diff
@@ -19,19 +19,21 @@ plugins {
 // The version is specified here to prevent an inconsistent version from being seen by different tasks.
 version = System.getenv("JAVA_PROTOS_VERSION") ?: "0.1.0-SNAPSHOT"
 
-val grpcProtobufVersion = "3.21.2"
-val grpcVersion = "1.47.0"
-val guavaVersion = "31.1-android"
+val grpcProtobufVersion = "3.22.2"
+val grpcVersion = "1.53.0"
 
 dependencies {
-    implementation("io.grpc:grpc-protobuf:$grpcVersion")
-    implementation("com.google.api.grpc:proto-google-common-protos:2.0.1")
+    api(platform("io.grpc:grpc-bom:$grpcVersion"))
+    api(platform("com.google.protobuf:protobuf-bom:$grpcProtobufVersion"))
 
-    implementation("com.google.protobuf:protobuf-java:${grpcProtobufVersion}")
-    implementation("com.google.guava:guava:${guavaVersion}")
+    api("io.grpc:grpc-stub")
+    api("io.grpc:grpc-api")
+    api("com.google.protobuf:protobuf-java")
+    api("com.google.guava:guava:31.1-android") // version pulled from protobuf-java
```

Comment:
So are you switching these from impl to `api` because there are types from them that appear in the APIs of the generated code?  And thus one can't consume the generated code without these deps?

Mostly asking because of concern about possibly conflicting versions of these between us and our users.

Reply:
Yep. grpc-stub, protobuf-java, and guava are required to use the apis. I don't think grpc-api is, so I'll change it. I really want a tool that lets me see every place a dependency is used in a project.

Reply:
The presence of guava does give me a bit of pause. I want to investigate shading this and many of the other dependencies in the java sdk.

Reply:
Do we want to leave them as implementation to force users to explicitly depend on them? The former submodule in the java sdk used api dependencies transitively before I made them explicit.